presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionWithIndexFunction.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.OperatorDependency;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.function.OperatorType.EQUAL;
+import static com.facebook.presto.util.Failures.internalError;
+
+@Description("Returns the position of the first occurrence of the given value in array (or 0 if not found)")
+@ScalarFunction("array_position")
+public final class ArrayPositionWithIndexFunction
+{
+    private ArrayPositionWithIndexFunction() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BIGINT)
+    public static long arrayPositionWithIndex(
```

Comment:
You can put the implementation in `ArrayPositionFunction.java`. No need to add a separate class.

Reply:
@rongrong are you saying I should just move the class declaration for `ArrayPositionWithIndexFunction` in `ArrayPositionWithIndexFunction.java` into `ArrayPositionFunction.java` (as a `final` instead of `public final` class)?

Or are you suggesting I move the public methods in the `ArrayPositionWithIndexFunction` class into the `ArrayPositionFunction` class? When I tried declaring these static methods in `ArrayPosition` earlier, I get an exception on my test relating to the signature of the implementation.
```
Caused by: java.lang.IllegalArgumentException: Implementations with type parameters must all have matching signatures. presto.default.array_position<T:comparable>(array(T),T,bigint):bigint does not match presto.default.array_position<T:comparable>(array(T),T):bigint
```

Reply:
Hey @rongrong! Just following up here. Any ideas on how I need to annotate the new methods with the additional argument inside `ArrayPositionFunction` to avoid this exception?

Reply:
Sorry I was wrong. The classes with multiple scalars does not support specialized implementation so it's easier to keep them in separate classes.